lib/puppet/type/file/content.rb

```diff
@@ -220,6 +221,19 @@ def get_from_source(source_or_content, &block)
       end
     end
 
+    def get_from_http_source(source, &block)
```

Comment:
ah, that makes sense. Extending the indirector to handle streaming results is definitely outside the scope of this PR